test/chemistry/test_oovqe.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test OOVQE """
+
+import unittest
+from test.aqua import QiskitAquaTestCase
+from ddt import ddt
+from qiskit.aqua import aqua_globals
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.components.optimizers import COBYLA
+from qiskit.chemistry.core import Hamiltonian
+from qiskit.chemistry.drivers import PySCFDriver, UnitsType
+from qiskit.chemistry.components.variational_forms import UCCSD
+from qiskit.chemistry.components.initial_states import HartreeFock
+from qiskit.chemistry.core import TransformationType, QubitMappingType
+from qiskit.chemistry.algorithms.minimum_eigen_solvers import OOVQE
+from qiskit.aqua.operators.expectations import MatrixExpectation
+from qiskit import Aer
+
+
+@ddt
+class TestOOVQE(QiskitAquaTestCase):
+    """ Test of the OOVQE algorithm"""
+
+    def setUp(self):
+        super().setUp()
+        self.energy_vqe = -2.77  # energy of the VQE with pUCCD ansatz and LBFGSB optimizer
+        self.initial_point = [0.039374, -0.47225463, -0.61891996, 0.02598386, 0.79045546,
```

Comment:
Why the "special" initial point here. I could imagine a test where it checked with setting a custom one. But its not tested with the default which I would think would be the most common way people use the algo.

Reply:
It is not possible to run the full OOVQE from default as it would take too much time for the unittest, therefore the initial point is given and the algorithm is checked for a few iterations to give known energies (better than VQE). If we use default we won't test anything as energies do not change enough to see that OOVQE improved on VQE, we would just add time on unittest for nothing.

Reply:
Is there some closer value then that perhaps can be used to do few iterations. max_iters of 1 for COBYLA just is one evaluation at the solution which seems like it would not do so much.

Reply:
Ah sorry, I think the problem is in why. I keep the max_iters = 1 for that test because I want to just do a single orbital rotation and energy evaluation because I know the energy I should get.  If we break the orbital rotation part of the code, this value will be wrong. For the other tests you can see that I have 3 or 2 as max_iters ( test_oovqe, test_oovqe_iterative) so optimizer does something at least to test that results are returned and the energy is somewhat improved.